src/starknet_storage/storage.rs

```diff
@@ -0,0 +1,98 @@
+use super::errors::storage_errors::StorageError;
+use cairo_rs::types::instruction::Res;
+use std::str;
+
+//* ------------------
+//*   Storage Trait
+//* ------------------
+
+/* -----------------------------------------------------------------------------------
+   -----------------------------------------------------------------------------------
+
+    This module implements the trait for the the storages operations.
+    The trait default functions handle the different data types that can be stored,
+    some assumptions and work arounds have been taken for this.
+
+    * All data types are turned into Vec<u8> before being passed to the set_value function.
+    This is due to rust restrictions on parameter types for the functions in traits.
+
+    * The same is true for get_value but this returns an Option because there may be nothing in the storage for a given key.
+
+    * float types are assumed to be f32, in case of needing to store f64, get_double and set_double functions can be implemented.
+
+    * Strings are assumed to be UTF-8 valid encoding, using a different format falls in an error.
+
+    * get_str returns a String type rather than a str to avoid lifetimes or reference issues.
+
+  -----------------------------------------------------------------------------------
+  -----------------------------------------------------------------------------------
+*/
+
+pub(crate) trait Storage {
+    type T;
```

Comment:
I don't see `T` being used anywhere.

Reply:
i must be deleted, it was going to be used as return types for functions but it brings too much issues.